### ISSUE TITLE
Enabling show bound boxes for 3D layer styling of LAZ files crashes QGIS

### ISSUE THREAD

**Issue by antoniolocandro** [qgis/QGIS#47409]
### What is the bug or the crash?

Enable show bounding box crashes QGIS

### Steps to reproduce the issue

1. load some laz files
2. Use Single color as 3D style
3.  Add a 3D Map view
4. Enable Show bound boxes
5.  Crash

![image](https://user-images.githubusercontent.com/4627073/154372931-8b8e2b1f-44fd-4876-bb06-29295a3a906d.png)

Stack trace
Python Stack Trace
Windows fatal exception: access violation

Thread 0x000072d8 (most recent call first):

Stack Trace
No stack trace is available.

QGIS Info
QGIS Version: 3.23.0-Master
QGIS code revision: b0a83e9460
Compiled against Qt: 5.15.2
Running against Qt: 5.15.2
Compiled against GDAL: 3.4.1
Running against GDAL: 3.4.1

System Info
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.22543

### Versions

QGIS Version: 3.23.0-Master
QGIS code revision: b0a83e9460
Compiled against Qt: 5.15.2
Running against Qt: 5.15.2
Compiled against GDAL: 3.4.1
Running against GDAL: 3.4.1

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

Dev version used on purpose, new profile also tested same results

**Comment by gioman** [comment]
> QGIS Version: 3.23.0-Master

@antoniolocandro can't replicate with my own data. Master on Windows.

**Comment by nicogodet** [comment]
@antoniolocandro Win 11 ?

**Comment by antoniolocandro** [comment]
@nicogodet yes, using dev version of Win 11 maybe related to that but as people start to migrate to it more maybe may encounter similar issues

**Comment by gioman** [comment]
@antoniolocandro it can still be data related, so please attach one of your datasets.

**Comment by antoniolocandro** [comment]
Here is some sample data

[data.zip](https://github.com/qgis/QGIS/files/8088945/data.zip)

**Comment by gioman** [comment]
> [data.zip](https://github.com/qgis/QGIS/files/8088945/data.zip)

still no crash on Win10, so it may be really a Win11 thing.
